cmd/thanos/compact.go

```diff
@@ -178,12 +179,12 @@ func runCompact(
 	g.Add(func() error {
 		defer runutil.CloseWithLogOnErr(logger, bkt, "bucket client")
 
-		if !wait {
+		if repeat == 0 {
 			return f()
 		}
 
-		// --wait=true is specified.
-		return runutil.Repeat(5*time.Minute, ctx.Done(), func() error {
+		// --repeat is specified.
+		return runutil.Repeat(backoff.NewConstantBackOff(repeat), ctx.Done(), func() error {
```

Comment:
I think you confused `Repeat` with `Retry` 

We don't want to have backoff here. This is just periodic compaction+downsampling+retention run that is periodically ran... This is executed for both retry one error, sure, but also for normat thing.

Reply:
Actually not, my thought was use the same interface to pass time intervals to Repeat or Retry methods. 

So further, we can pass custom backoff, not only constant, but the customizable.
But the term of backoff seems not to be clear, indeed.

Reply:
Yup, got confused. ^^ Constant backoff like a  "hot ice"  or "moving constant" opposite words.

Also not sure any backoff is nice for repeat, ticker based function. Maybe we need to fix just naming here

Reply:
fixing naming of backoff? that's a external package.
I can't leave it as it was to avoid confusion with backoff vs repeat (not retry)

Reply:
yup